go/registry/api/api.go

```diff
@@ -505,6 +506,46 @@ func VerifyRegisterNodeArgs( // nolint: gocyclo
 		return nil, err
 	}
 
+	// Make sure that the consensus and P2P keys, as well as the
+	// committee certificate are unique (between themselves and compared to
+	// other nodes).
+	if n.Consensus.ID.Equal(n.P2P.ID) {
+		logger.Error("RegisterNode: node consensus and P2P IDs must differ",
+			"node", n,
+		)
+		return nil, ErrInvalidArgument
+	}
+	for _, rn := range regNodes {
+		if rn.ID.Equal(n.ID) {
+			// Node descriptor is being updated, IDs below are not duplicates.
+			continue
+		}
+
+		if rn.Consensus.ID.Equal(n.Consensus.ID) {
+			logger.Error("RegisterNode: duplicate node consensus ID",
+				"node", n,
+				"existing_node", rn,
+			)
+			return nil, ErrInvalidArgument
+		}
+
+		if rn.P2P.ID.Equal(n.P2P.ID) {
```

Comment:
note that currently if multiple validators would use the same sentry node, they would register with same p2p ids

Reply:
That will no longer work after this PR is merged.  I think the idea of issue #2421 was to discourage sharing of keys.

Reply:
no keys are shared, the nodes just use the same sentry node as a proxy (and register with sentries id and address, the way it's currently implemented). should probably explicitly enforce one validator per sentry if we plan to not support it anymore.

Reply:
nevermind, it will be fine actually, as the `node.Consensus.Address.ID` will be duplicated, not the `node.Consensus.ID`, so it should be fine :+1: